src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -22,76 +28,325 @@
+                    This.CastTo(iModelTInterface).Invoke(nameof(IPersistableModel<object>.Write), options)
+                );
+
+                // object IPersistableModel<object>.Create(BinaryData data, ModelReaderWriterOptions options)
+                yield return new
+                (
+                    new MethodSignature(nameof(IPersistableModel<object>.Create), null, null, MethodSignatureModifiers.None, typeof(object), null, new[] { KnownParameters.Serializations.Data, KnownParameters.Serializations.Options }, ExplicitInterface: iModelObjectInterface),
+                    // => (IPersistableModel<T>this).Read(options);
+                    This.CastTo(iModelTInterface).Invoke(nameof(IPersistableModel<object>.Create), data, options)
+                );
+
+                // ModelReaderWriterFormat IPersistableModel<object>.GetFormatFromOptions(ModelReaderWriterOptions options)
+                yield return new
+                (
+                    new MethodSignature(nameof(IPersistableModel<object>.GetFormatFromOptions), null, null, MethodSignatureModifiers.None, typeof(string), null, new[] { KnownParameters.Serializations.Options }, ExplicitInterface: iModelObjectInterface),
+                    // => (IPersistableModel<T>this).GetFormatFromOptions(options);
+                    This.CastTo(iModelTInterface).Invoke(nameof(IPersistableModel<object>.GetFormatFromOptions), options)
+                );
+            }
+
+            static IEnumerable<MethodBodyStatement> BuildModelWriteMethodBody(JsonObjectSerialization? json, XmlObjectSerialization? xml, ModelReaderWriterOptionsExpression options, CSharpType iModelTInterface)
+            {
+                // var format = options.Format == "W" ? GetFormatFromOptions(options) : options.Format;
+                yield return Serializations.GetConcreteFormat(options, iModelTInterface, out var format);
+
+                yield return EmptyLine;
+
+                var switchStatement = new SwitchStatement(format);
+
+                if (json != null)
+                {
+                    var jsonCase = new SwitchCase(Serializations.JsonFormat,
+                        Return(new InvokeStaticMethodExpression(typeof(ModelReaderWriter), nameof(ModelReaderWriter.Write), new[] { This, options }))
+                        );
+                    switchStatement.Add(jsonCase);
+                }
+
+                if (xml != null)
+                {
+                    /*  using MemoryStream stream = new MemoryStream();
+                        using XmlWriter writer = XmlWriter.Create(stream);
+                        ((IXmlSerializable)this).Write(writer, null);
+                        writer.Flush();
+                        // in the implementation of MemoryStream, `stream.Position` could never exceed `int.MaxValue`, therefore this if is redundant, we just need to keep the else branch
+                        //if (stream.Position > int.MaxValue)
```

Comment:
Update the comment as well

Reply:
well I intentionally kept them and changed those to comments (with more comments)